src/hooks/useCardAnalytics.ts

```diff
@@ -28,14 +29,17 @@ function isDirectAnswer(data: unknown): data is DirectAnswerData {
     (data as DirectAnswerData)?.type === DirectAnswerType.FieldValue;
 }
 
-export function useCardAnalytics(): (
-  cardResult: Result | DirectAnswerData, analyticsEventType: CardAnalyticsType
+export function useCardAnalytics<T = DefaultRawDataType>(): (
+  cardResult: Result<T> | DirectAnswerData, analyticsEventType: CardAnalyticsType
 ) => void {
   const analytics = useAnalytics();
   const verticalKey = useSearchState(state => state.vertical.verticalKey);
   const queryId = useSearchState(state => state.query.queryId);
 
-  const reportCtaEvent = useCallback((result: DirectAnswerData | Result, eventType: CardCtaEventType) => {
+  const reportCtaEvent = useCallback((
+    result: DirectAnswerData | Result<T>,
```

Comment:
can this union type be defined somewhere so we don't have to repeat it in multiple places?

Reply:
sure